src/mapping.ts

```diff
@@ -9,170 +9,158 @@ import {
   const tokenId = event.params.tokenId.toString();
+  let collection = Collection.load(MERGE_EXTADDR);
+
+  if (!collection) {
+    collection = createCollection(MERGE_EXTADDR); // NEW for collection entity - first instance of minting
+  }
 
-  // Step 0 && Step 1: set NG_OMNIBUS as a whitelisted address as it is done in the constructor, and initialize new users
   if (!user) {
     user = new User(to);
-    if (NG_OMNIBUS) {
+    collection.totalUniqueOwnerAddresses++;
+    collection.save();
+    user.save();
+    if (to == NG_OMNIBUS) {
       user.whitelist = true;
     }
   }
 
-  user.save();
-
-  let nft: NFT;
-
-  // Step 2a: either create new NFT if in minting phase
+  let scenario: string;
   if (from == EMPTY_ADDRESS && to == NG_OMNIBUS) {
-    nft = createNFT(event, user.id);
-  } else {
-    // Step 2b: OR updateNFT for all other Transfer scenarios incl: temporary transfers right before merging/burning, simple burning (direct to dead address, which then transfers to address(0))
-    nft = updateNFT(tokenId, user.id);
+    createNFT(event, user.id);
+    scenario = "Mint";
+    updateCollection(tokenId, scenario);
+  } else if (to == EMPTY_ADDRESS && from == NG_OMNIBUS) {
+    scenario = "NiftyBurn";
+    updateNFT(tokenId, user.id);
+    updateCollection(tokenId, scenario);
+  } else if (to == EMPTY_ADDRESS) {
+    scenario = "NonNiftyBurn";
+    updateNFT(tokenId, user.id);
+    updateCollection(tokenId, scenario);
+  } else if (from == ADDR_DEAD) {
+    scenario = "StraightBurn";
+    updateCollection(tokenId, scenario);
+  } else if (!(from == NG_OMNIBUS && to != EMPTY_ADDRESS && to != ADDR_DEAD)) {
+    scenario = "normTransfer"; //transfer to someone who doesn't own a mergeNFT yet
+    updateNFT(tokenId, user.id);
+    collection.totalUniqueOwnerAddresses--; // to counterbalance the ++ when making this new user entity.
+    collection.save();
   }
 
-  nft.save();
+  user.save();
 }
 
 /**
  * @notice updates when new alphaMass identified
  * @param event AlphaMassUpdate
- * @dev TODO: need to handle the old alphaMass still. So if one tokenId is alpha one day, but the next a different one is, they will both have the 'isAlpha' field as 'true'
+ * @dev TODO: need to handle the old alphaMass still - tbd in update 4 PR. So if one tokenId is alpha one day, but the next a different one is, they will both have the 'isAlpha' field as 'true'
  */
 export function handleAlphaMassUpdate(event: AlphaMassUpdate): void {
   const alphaTokenId = event.params.tokenId.toString();
-  const nft = NFT.load(alphaTokenId) as NFT;
-
-  nft.mass = event.params.alphaMass;
-  nft.isAlpha = true;
-  nft.color = checkColor(nft.value.toString(), nft.isAlpha);
-
-  nft.save();
+  const nft = NFT.load(alphaTokenId);
+
+  // TODO: remove this possibly: nft should exist, but doing this since compiler was giving me issues with null
+  if (!nft) {
+    const tokenIdString = event.params.tokenId.toString();
+    const tokenId = event.params.tokenId;
```

Comment:
these two lines are kinda useless because each variable only gets used once.

you can just do 
```
    const nftValue = contract.getValueOf(event.params.tokenId);
   ```

etc.